ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -1625,55 +1503,28 @@ const RuntimePanel = fp.flow(
         status as RuntimeStatus
       );
 
-    const initialRuntimeConfig = {
-      computeType: initialCompute,
-      machine: initialMasterMachine,
-      diskSize,
-      dataprocConfig,
-      gpuConfig,
-      pdSize,
-      autopauseThreshold: initialAutopauseThreshold,
-    };
-
-    const newRuntimeConfig = {
-      computeType: selectedCompute,
-      machine: selectedMachine,
-      diskSize: selectedDiskSize,
-      dataprocConfig: selectedDataprocConfig,
-      gpuConfig:
-        selectedCompute === ComputeType.Standard ? selectedGpuConfig : null,
-      pdSize: selectedPdSize,
-      autopauseThreshold: selectedAutopauseThreshold,
-    };
-
-    const gceExists = runtimeExists && initialCompute === ComputeType.Standard;
-    const dataprocExists = dataprocConfig !== null;
-    const enablePD =
-      serverConfigStore.get().config.enablePersistentDisk &&
-      (pdExists || !gceExists);
-    const unattachedPdExists = enablePD && !gceExists && pdExists;
-    const pdSizeReduced = selectedPdSize < pdSize;
-    // A runtime context can wrap/pass complex runtime context and also make the code cleaner
-    const runtimeCtx = {
-      runtimeExists: runtimeExists,
-      gceExists: gceExists,
-      dataprocExists: dataprocExists,
-      pdExists: pdExists,
-      // Here the enablePD is not simply the pd feature flag.
-      // It stands for the point when user has no old version gce instances exists and the pd feature flag is also True at the same time.
-      // By using this predicate, we can differentiate the old version and pd version panel much more easily. The code is also cleaner.
-      // In addition, after all the old users change to the new pd version, we can simply replace all occurrences of this variable to True.
-      // The code refactor cost is low.
-      enablePD: enablePD,
-      unattachedPdExists: unattachedPdExists,
-    };
-    const runtimeDiffs = getRuntimeConfigDiffs(
-      initialRuntimeConfig,
-      newRuntimeConfig,
-      runtimeCtx
-    );
-    const updateMessaging = diffsToUpdateMessaging(runtimeDiffs);
-    const runtimeChanged = runtimeExists && runtimeDiffs.length > 0;
+    const dataprocExists =
+      runtimeExists && existingRuntimeConfig.dataprocConfig !== null;
+
+    const attachedPdExists =
+      !!persistentDisk && runtimeExists && existingRuntimeConfig.detachableDisk;
+    const unattachedPdExists = !!persistentDisk && !attachedPdExists;
```

Comment:
runtime exists, but runtime does not have detachableDisk. This should be false.

Reply:
I don't see any issue with these lines. There are three inputs:

- pd exists
- runtime exists
- runtime has an attached PD

Your example didn't specify whether PD exists, so we can go through both scenarios:

1. PD exists, runtime exists, runtime does not have PD: **true** is correctly returned; there is a PD, and it is unattached
2. PD does not exist: **false** is correctly returned

Reply:
I was referering to PD exists, runtime exists, `runtimeExists=false`, `existingRuntimeConfig.detachableDisk = true`.
But it might not be possible if `runtimeExists` is the pre-requirement for `existingRuntimeConfig.detachableDisk`

Reply:
Yes, detachableDisk setting doesn't matter if the runtime does not exist